Source/JavaScriptCore/runtime/VMTraps.h

```diff
@@ -293,8 +295,9 @@ class VMTraps {
         m_trapBits.exchangeAnd(~event);
     }
 
-    JS_EXPORT_PRIVATE CONCURRENT_SAFE void cancelThreadStopIfNeeded();
-    JS_EXPORT_PRIVATE CONCURRENT_SAFE void requestThreadStopIfNeeded(Event);
+    CONCURRENT_SAFE void cancelThreadStopIfNeeded() WTF_REQUIRES_LOCK(m_trapSignalingLock);
+    CONCURRENT_SAFE void requestThreadStopIfNeeded(Locker<Lock>&) WTF_REQUIRES_LOCK(m_trapSignalingLock);
+    CONCURRENT_SAFE void updateThreadStateIfNeeded();
```

Comment:
`updateThreadStateIfNeeded()` needs to be `JS_EXPORT_PRIVATE` because it is called from inline functions.  The other 2 does not need to be because they are no longer called from inline functions, just from `updateThreadStateIfNeeded`.

nit: Please also rename `updateThreadStateIfNeeded` to `updateThreadStopStateIfNeeded` or `updateThreadStopMachineryIfNeeded` or just `updateThreadStopIfNeeded` (to be consistent with `cancelThreadStopIfNeeded` and `requestThreadStopIfNeeded`).  I think `updateThreadStateIfNeeded` is too general and implies that it modifies other types of thread state that it does not (e.g. native thread suspension, or register / stack state).